core/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -197,7 +197,7 @@ public final ExprEval castTo(ExprType castTo)
           return this;
         case LONG:
           if (value == null) {
-            return ExprEval.of(null);
+            return ExprEval.ofLong(null);
```

Comment:
is it possible to add a test?

Reply:
What do you think about moving ExprEval.of(long) and ExprEval.ofLong(Number) to LongExprEval and calling them LongExprEval.of and LongExprEval.ofBoxed? It might make the code easier to understand and less prone to programmer error.

Reply:
I'm going to merge this fix now in the interests of getting it backported to 0.13.0. But, please consider the above comments when making future changes to this file.